.. _first-week:
First Week
==========
In the first week, you will see, separate and determine oribatid mites and process them in the molecular lab to obtain DNA sequences for phylogenetic analyses.

Goals
--------------
.. note::
	**You will…**
	  - gain basic knowledge of oribatid mites’ ecology, evolution, phylogeny and body plans.
	  - realize that their body plan is often linked with their taxonomy and phylogeny.
	  - will understand how to distinguish oribatid mites from other soil-fauna and other mites.
	  - will be able to recognize the most relevant and useful morphological characters to separate oribatid mites into major groups.
	  - be able to determine oribatid mites to genus, if possible even to species level, using a determination key.
	  - be able to organise your collected and determined material within a team, which includes labelling and keeping a common list updated.
	  - learn how DNA extraction works step by step and you will be able to tell properties of the most relevant chemical substances that are essential for DNA extraction.
	  - be able to explain how a PCR works and how to visualize the results.
	  - be able to tell the differences between the different genes (markers) we will use for the PCRs.
	  - understand how Sanger sequencing works, why we are using it in this course and how to prepare samples for Sanger sequencing.

Monday
------
You already know, how to collect soil animals. In the morning, we start with a general introduction to the soil fauna. In the afternoon, we work with oribatid mites only. And at the end of the day, you have a collection of different mite species, roughly divided into distinct morphological types (morphotypes).
In the morning, you will have a general introduction into soil animal systematics and meet the most common representatives to make you familiar with the term “mesofauna”.
You will also receive an introduction to the taxonomy and phylogeny of oribatid mites and learn more about their characteristic body plans and lifestyles, which will help you to distinguish oribatid mites from other mites.
In the afternoon, you will see the soil fauna with your own eyes through a stereomicroscope. While many of the soil animals will be dead in your samples, all oribatid mites and other taxa, will be alive. Oribatid mites are a very diverse group, ranging from fast runners to very slow movers; from tiny transparent ones to large, pitch-black species. Some species are smooth as glass, others are covered in dirt. However, all oribatid mites have in common an extensive arsenal of defense strategies.

.. tip::
   Take your time to observe oribatid mites and their characteristic movements and bodies. Write down three to five features (morphological and/or behavioural) of three living oribatid mites you observe. Remember, a character is everything you can observe. What type of defense strategy do the different oribatids display? Save your notes, we will use them later in this course. It may be helpful to make drawings or sketches of the observed species to more easily distinguish morphotypes based on certain character(s).

.. note::
  **At the end of the day, you will be able to recognize the most relevant morphological characters to:**
    - separate oribatid mites from other mites
    - sort oribatid mites into major groups
    - to sort oribatid mite species into groups of morphotypes
    - determine oribatid mites to genus, and maybe to species level.

Tasks of the Day
^^^^^^^^^^^^^^^^
.. important::
	  - Sort oribatid mites by morphotypes. You can collect over 10 different morphotypes, at least two are required (use glass dishes filled with ethanol).
	  - At the end of the day, transfer each morphogroup (all individuals belonging to the same morphotype) into a separate vial.
	  - Transfer one or two individuals (depending on your total number) in a glass vial and add a small label.
	  - The remaining individuals (up to eight or more, at least one ore two are required depending on your total number) go into 2 ml tubes. Label them in the same way as the samples in the glass vials.
	  - Write down three to five characters (morphological and/or behavioural) of three oribatid mite species. Keep your notes, we will be using them later in this course.

Feedback Monday
^^^^^^^^^^^^^^^
To provide feedback, please complete our questionnaire: [Questionnaire Link](https://www.example.com/questionnaire).

.. tip::
   Just in case, you can watch the :ref:`lectures` again at any time.


Tuesday
-------
Goals
^^^^^
Yesterday, you collected oribatid mites that were slowly cooking in lactic acid over night to brighten them for better examination under the microscope. Without this treatment, it would be impossible to recognize important morphological features.

.. note::
  **Today, you will…**
    - deepen your knowledge about how to determine oribatid mites.
    - learn to recognise additional morphological characters.
    - be able to determine most of your specimen to species level.

For the species level identification (using the microscope), please remove one or two individuals from the glass vials; the remaining specimen (in 2 ml reaction tubes) are used for molecular work.

Summary
^^^^^^^
Yesterday, you gained an overview about soil-biodiversity and the most common mesofauna representatives. Today we got a deeper insight into the fascinating world of body shapes and forms that are so characteristic of oribatid mites. You are also familiar with the 'morphotype' concept and able to recognize families and genera of oribatid mites, in some case down to to the species level, by form, shape, and individual characters.
Today your task is to determine at the species level the morphotypes you collected yesterday (if you didn't do so yesterday). Use only one individual from your morphotype collection and leave the others untouched. Please enter all species you have identified in a common taxa list, including the number of morphotypes.
At the end of the day, the entire class will have a shared collection of oribatid mite species of specific morphotypes; you learned how to label and organise samples and intermediate results (i.e., your shared taxa list) in a transparent and unique way.

Tasks of the Day
^^^^^^^^^^^^^^^^^
.. important::
    1. Please enter the sample number, species name, morphotype number, and your name in the common taxa list.
    2. At the end of the day, everyone should have five different species identified.

Question of the Day
^^^^^^^^^^^^^^^^^^^
.. important::
    Why will we use the undetermined morphospecies for molecular work?

Feedback Tuesday
^^^^^^^^^^^^^^^^
To provide feedback, please complete our questionnaire: [Questionnaire Link](https://www.example.com/questionnaire).

.. tip::
   Just in case, you can watch the :ref:`lectures` again at any time.


Wednesday
---------
Summary
^^^^^^^
We start the day with revising what you have learned so far. By today you are able to recognize oribatid mites in general and to sort morphotypes into the six major groups (Palaeosomata, Parhyposomata, Enarthronota, Mixonomata, Desmonomata, and Brachypylina). You can name a handful of characters that are typical for each group and can assign at least one species to at least four of the six major groups. You know where oribatid mites live, and you have a general understanding of their ecology and phylogeny.
After working with oribatid mites on a morphological basis, you now start to work with them on a molecular level. You will get familiar with the general process of DNA extraction and laboratory safety rules. After you successfully extracted and purified your DNA, you start a PCR reaction that is going to run overnight. The DNA extraction includes a long incubation step for ~2 hours. Be prepared to work until after 5:00 pm today.
You learn the principles of DNA extractions and PCR and the different steps that are part of these methods. You will get to know the relevant reagents and what they do.
At the end of the day, you have purified DNA stored in the freezer and a PCR running.

Goals
^^^^^
In this course, we use a DNA extraction method based on spin columns with a silica membrane, provided as a DNA extraction kit. 'Kit' means, that buffers, reaction tubes, incubation, and centrifugation procedures are delicately matched, ensuring easy processing with an optimal outcome (i.e. DNA yield).

.. note::
  **You will…**
    - learn how to work safe and clean in a molecular lab.
    - learn how to use a microlitre pipette.
    - learn how to handle your samples without contaminating the lab work space, your samples or yourself.
    - learn the names of the most relevant chemical substances and conditions for the different steps during DNA extraction.
    - learn how to set up a PCR efficiently.
    - learn what a PCR mastermix is.
    - learn how a PCR reaction works and what a thermo cycler does.

Tasks of the Day
^^^^^^^^^^^^^^^^
.. important::
    1. Do DNA extractions and PCR reactions with your samples.
    2. Work clean and tidy in the lab and obey the lab rules.

Questions of the Day
^^^^^^^^^^^^^^^^^^^
.. important::
  1. What is SDS and what is it doing with your samples?
  2. Explain the difference between mechanical and biochemical processess used to destroy animal tissue?
  3. Why do you have to purify your samples?
  4. Why is it important to have ethanol-free DNA ath the end of the DNA extraction process?
  5. What is the purpose of the silica membrane?
  6. Why is pH relevant for DNA extraction? Which pH values are most relevant here?
  7. What is a chaotropic salt and what is it used for in DNA extraction?
  8. What did you observe when working with ATL (lysis) buffer and AW1 buffer (wash buffer 1?
  9. What does a PCR mastermix contain?
  10. What is difference between the various PCR mastermixes you have used?
  11. Is it possible to combine different PCR primers in one PCR reaction?
  12. Which genes did you use for your PCR reaction(s)?
  13. Why is the lid of the thermal cycler heated to 100°C? 🥵
  14. What is a hot start Taq polymerase (= HotStarTaq)?
  15. What do you have to consider when working with a HotStarTaq?

Feedback Wednesday
^^^^^^^^^^^^^^^^^^
To provide feedback, please complete our questionnaire: [Questionnaire Link](https://www.example.com/questionnaire).

.. tip::
   Just in case, you can watch the :ref:`lectures` again at any time.


Thursday
--------
You have been introduced to the basic and most common methods in molecular work i.e., DNA extraction and PCR. Today you will perform gel electrophoresis to visualize the products of your PCRs. Successful PCR reactions are then purified and sent for sequencing.
At the end of the day, you have completed the wet-lab part of your molecular work. You know the necessary steps to obtain DNA sequences and have a semi-professional understanding of the steps involved, from DNA extraction to purification of the PCR product.
The purified PCR products are sent to a sequencing company that uses the standard Sanger sequencing method.

Goals
^^^^^
Today, you pour an agarose gel and load a small amount of your PCR product onto the gel to check whether your PCR worked or not. Before loading your samples onto the gel, be sure to add loading dye. After checking the gel, purify (i.e., wash) the PCR products with bright and clear bands in order to send them for sequencing.

Questions of the Day
^^^^^^^^^^^^^^^^^^^
.. important::
  1. What is agarose and why is it used to visualize DNA/PCR products?
  2. What is a loading dye, what is it used for?
  3. Why is it important to add a standard size ladder onto your gel?
  4. What happens if you change the agarose concentration of your gel from 1% and a 2%?
  5. Why is it important to use the same buffer for melting and running your gel?
  6. Why do we use a buffer instead of water?
  7. Why do we visualize our DNA/PCR products?
  8. Why is it necessary to purify positive PCR products before sequencing?
  9. When washing your PCR products, the first buffer will be yellow in color. Why? What does it indicate?

Feedback Thursday
^^^^^^^^^^^^^^^^^
To provide feedback, please complete our questionnaire: [Questionnaire Link](https://www.example.com/questionnaire).


Friday
-------
Homework
^^^^^^^^
You can start a Discussion group in the `Forum <https://studip.uni-goettingen.de/dispatch.php/course/forum/index/index/ee59763e66f3b71b69c917f633cb857d?cid=d7b80997f5efda59609a4cf69a04dbf7#ee59763e66f3b71b69c917f633cb857d>`_ to work on the questions and tasks of the past days and for today.

Tasks of the Day
^^^^^^^^^^^^^^^^
Watch the following YouTube videos for a general introduction. Get familiar with the steps involved, the essential idea of fluorescence base Sanger DNA sequencing, and the advantages of capillary electorophoreses.

.. youtube:: YeHtjO7vlyg
.. youtube:: CN21sVuXkkw
.. youtube:: k0ZSOR1Gd1M
.. youtube:: Wpww8bb63zU
.. youtube:: KTstRrDTmWI
.. youtube:: FvHRio1yyhQ

.. important::
  Make sure you are able to answer the following questions. Help yourself online…
   - Can you draw the structure of a DNA and a RNA molecule with following sequence A-T-G?  There is no need to know the structure of single bases, abbreviate them with A, C and G. However, you need to know the difference of DNA and RNA molecules and their structure.
   - What is ddNTP and what is the difference to dNTP?
   - What are the main advantages of the modern fluorescence based Sanger DNA sequencing approach compared to the original (radio-labelled) one.
   - What happens to your PCR primers during Sanger sequencing? Are they even used?
   - Why do we sequence the forward and reverse strands of the PCR product separately? Is it important for the method itself or just helpful for analyzing our data?

Watch 'A brief tour of Geneious Prime <https://go.geneious.com/video/getting-started-in-geneious-prime>`_.
If you are going to use your own PC on Monday next week, install the `30-days trial version of Geneious Prime <https://manage.geneious.com/free-trial>`_.

.. note::
  As soon as possible, but latest on Monday next week, you will receive the raw reads of your sequences. Your PCR products were sequenced in forward and reverse direction, meaning that two raw reads form a consensus sequence. In order to obtain a consensus sequence, assemble the pairs of raw reads and – **very important!** – check for any sequencing errors and/or ambiguous positions. At the end, we want to have good to high quality consensus sequences for downstream analyses.

.. important:: 
  Watch the tutorial about `Sanger Sequencing <https://www.geneious.com/academy/getting-started/?wvideo=0htkkxq986>`_ on the Geneious Academy webpage. It will explain the most important and common terms such as sequence assembly, sequence checking, ambiguous positions, consensus sequence, etc, and explains how to analyze your Sanger sequences.

Task preview for Monday morning, Second Week:
---------------------------------------------

Each of you will process a subsample of raw reads to assemble good to high quality consensus sequences. An assignment list will be provided in StudIP. 
Checkout `the **raw_sequences** directory on StudIP  <https://studip.uni-goettingen.de/dispatch.php/course/files/index/660b809717ef7407f199fdb5f7a87d89?cid=d7b80997f5efda59609a4cf69a04dbf7>`_


.. attention::
 Sequences must to be assembled, checked, and exported as consensus sequence files. Please upload them to StudIP by Monday 5:00 pm. Each consensus file you upload must contain a sample ID and your initials.

.. note::
  If you want, you can also start the Monday tutorials and tasks this Friday. Note: They need to be finished by Monday so we can stick to the Tuesday schedule. You are very welcome to ask questions in BBB from 11-12 am and 1-2 pm and we would be happy to support you in assembling, checking, exporting, and naming your sequences. You can also use the Forum to ask questions and discuss them with the other participants.

.. hint::
  Be aware: The sequencing facility may work quickly, so raw sequences may be available as early as Saturday/Sunday. If so, the assignment list will be provided at the weekend. If the tutorial is clear enough for you and raw sequences were uploaded on the weekend (please check yourself), you can start working on the weekend and upload the consensus sequences. In any case, consensus sequences must be uploaded to StudIP by 5:00 p.m. on Monday or any time earlier.
