.. _second-week:
Second Week
===========
Goals
-----
This week it's all about DNA sequence: How to work with them, where to get them from (if you do not wish to sequence animals yourself), how to infer phylogenetic trees.

All steps for handling sequence data and phylogenetic analyses are explained step by step in online tutorials. Everyday, a tutorial is dedicated to a different method. You must finish the respective tutorial and its exercises in order to be able to continue with the next tutorial on the next day.

.. note::
   However, you start this week with assembling, checking, and exporting your raw sequences (= sequencing results of your PCR products from last week) to obtain good or even high quality consensus sequences.  We start with your DNA sequences so that you will get familiar with…
  - Sanger DNA sequencing
  - sequence evaluation (or 'What's the difference between a bad and good DNA sequence?')
  - ambiguous (wobble) DNA positions and where do they come from
  - DNA sequences from public data repositories

 After you processed your own sequences, you can start with the online tutorial.

 .. note::
  At the end of the week you will know…
  - what a sequence file is
  - what a fasta file is
  - how to use public gene databases
  - how to edit sequences
  - how to check if your own sequencing results are correct
  - what  an multiple sequence alignment is
  - how to distinguish multifasta files from alignment files
  - what a model of sequence evolution is and why they are important for phylogenetic analyses
  - what the difference of Cluster methods and search algorithms is when constructing phylogenetic trees
  - what ML and BI means
  - how to do basic phylogenetics in R

Monday
------
Summary
^^^^^^^
Today we will start with recapitulating the molecular work of last week and discuss the method of Sanger Sequencing.
After that, you start with processing your sequencing results i.e., you will assemble, check and correct the raw reads that have been assigned to you (see assignment list) in Geneious Prime and export the respective consensus sequences.
Then, you can start with the online tutorial, which introduces you to the handling of sequence data and how to use Genbank, a public sequence data repository.
By doing the exercises in this tutorial, you will generate a toy dataset, which you will be using for the whole week. All following tutorials and exercises are based on this toy dataset.
The basic idea is, that all of you work with the same toy dataset, which makes it easier to compare results. However, it is also fine if you add some of your own sequences (those you checked and exported earlier today).

Tasks of the Day
^^^^^^^^^^^^^^^^
Task 1
""""""
.. note::
  1. Read the section :ref:`Geneious_Prime`
  2. Download and install the `30-days trial version of Geneious Prime <https://manage.geneious.com/free-trial>`_

