.. _third-week:
Third Week
===========

Today, it's all about sequence alignments and their importance for analysing genetic data. In this tutorial, you do sequence alignments with your toy datasets using the software `ClustalW <http://www.clustal.org/clustal2/>`_ implemented in :re:`BioEdit`, the sequence editor you used yesterday.
It is important to remember that sequence files, whether aligned or not, can be saved in different file formats. Input file formats can change between used software. If the format is not correct, the software isn't do anything for you. Knowing what the input file format should look like will help you overcome the initial hurdles when using phylogenetic software.

.. note::
  **At the end of the day, you know…**

  - how an alignment is generated by the Needleman-Wunsch algorithm
  - how computer algorithms (basically) perform
  - the meaning of penalty values and their effects on alignments
  - how to find criteria that will help you to decide if an alignment is good or not
  - the difference between sequence file formats, and the difference between multifasta and alignment files and how to recognize them

.. important::
  **Additionally, you need to know…**

  - the consequences of using coding versus non-coding sequences for an alignment
  - the meaning and use of reading frames when aligning your data

The different properties of coding and non-coding sequences will not be explained explicitly and we assume that you already know what reading frames are. However, if you are lost, do not hesitate to ask one of the tutors or me.

.. tip::
  Start a discussion in the Forum and/or write a Wiki-entry, explaining these issues to everyone.

